src/module/utils/links.ts

```diff
@@ -1,29 +1,82 @@
+import { SR5Actor } from "../actor/SR5Actor";
+import { SR5Item } from "../item/SR5Item";
+
 /**
  * Utils used for opening links
  */
 
 export class LinksHelpers {
+    /**
+     * Determine if the given string contains a PDF pattern.
+     * 
+     * @param candidate The string that might contain a PDF pattern like SR5 123
+     */
+    static isPDF(candidate: string | undefined): boolean {
+        if (!candidate) return false;
+        return candidate.split(' ').length === 2;
+    }
     /**
      * Determine if given string contains a url pattern.
      * 
+     * Parsing an url is expensive and doing so on UUIDs for JournalEntryPages will kill the browser :)
+     * Therefore we assume what is not anything else, might be a url.
+     * 
      * @param candidate The string that might contain a url
      * @returns true, when candidate contains a url pattern
      */
-    static isURL(candidate: string|undefined): boolean {
+    static isURL(candidate: string | undefined): boolean {
         if (!candidate) return false;
-        var urlRegex = '^(?!mailto:)(?:(?:http|https|ftp)://)?(?:\\S+(?::\\S*)?@)?(?:(?:(?:[1-9]\\d?|1\\d\\d|2[01]\\d|22[0-3])(?:\\.(?:1?\\d{1,2}|2[0-4]\\d|25[0-5])){2}(?:\\.(?:[0-9]\\d?|1\\d\\d|2[0-4]\\d|25[0-4]))|(?:(?:[a-z\\u00a1-\\uffff0-9]+-?)*[a-z\\u00a1-\\uffff0-9]+)(?:\\.(?:[a-z\\u00a1-\\uffff0-9]+-?)*[a-z\\u00a1-\\uffff0-9]+)*(?:\\.(?:[a-z\\u00a1-\\uffff]{2,})))|localhost)(?::\\d{2,5})?(?:(/|\\?|#)[^\\s]*)?$';
-        var url = new RegExp(urlRegex, 'i');
-        return url.test(candidate);
+
+        // Start in order of little performance expence
+        if (LinksHelpers.isPDF(candidate)) return false;
+        if (LinksHelpers.isUuid(candidate)) return false;
+
+        return true;
+    }
+    /**
+     * Determine if given string contains a valid uuid pattern.
+     * 
+     * FoundryVTT doesn't provide a method to check if a string is a valid uuid.
+     * We assume a uuid will end on a 16 digit id and will contain at least one dot.
+     * 
+     * @param candidate A string containing a Document.uuid
+     * @returns true, when candidate contains a valid uuid pattern
+     */
+    static isUuid(candidate: string | undefined) {
+        if (!candidate) return false;
+
+        const parts = candidate.split('.');
+        if (parts.length < 2) return false;
+        const idPart = parts.pop();
+        if (idPart?.length !== 16) return false;
```

Comment:
This assumption is based on uuid always going Document.Id even when nested. And as far as I know, Foundry uses 16 digit ids. This might be prone breaking, should Foundry change that. I'd prefer a Foundry internal way, though I haven't found one (also haven't looked all that long).

Reply:
Thats the Foundry Way-to-go.